libminifi/src/controllers/SSLContextService.cpp

```diff
@@ -80,7 +82,16 @@ bool SSLContextService::configure_ssl_context(SSL_CTX *ctx) {
         X509_free(cert);
         return false;
       }
-      if (SSL_CTX_use_PrivateKey(ctx, pkey) != 1) {
+	  while (ca != nullptr && sk_X509_num(ca)) {
+		  X509 * cacert = sk_X509_pop(ca);
+		  if (SSL_CTX_add_extra_chain_cert(ctx, cacert) != 1) {
+			  logging::LOG_ERROR(logger_) << "Failed to set additional certificate from  " << certificate << ", " << getLatestOpenSSLErrorString();
+			  EVP_PKEY_free(pkey);
+			  X509_free(cacert);
+			  return false;

```

Comment:
So we iterate through "ca" and free one of the certs in case of error and leave the rest. This definitely feels wrong.

Reply:
Fixed this with scope guard as well